File: Springboot/FinalAssessmentSpringBoot/src/main/java/com/example/FinalAssessmentSpringBoot/controllers/DishController.java
package com.example.FinalAssessmentSpringBoot.controllers;

import com.example.FinalAssessmentSpringBoot.beans.ResponseHandler;
import com.example.FinalAssessmentSpringBoot.models.Dish;
import com.example.FinalAssessmentSpringBoot.services.DishService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/dishes")
public class DishController {

    @Autowired
    private DishService dishService;

    @GetMapping("/")
    public ResponseEntity<Object> getAll(){
        return ResponseHandler.createResponse("Found dishes", HttpStatus.OK, dishService.getAll());
    }


    @GetMapping("/{id}")
    public ResponseEntity<Object> getDish(@PathVariable Long id){
        return ResponseHandler.createResponse("Found the dish", HttpStatus.OK, dishService.get(id));
    }

    @PostMapping("/")

    public ResponseEntity<Object> create(@RequestBody Dish dish) {
        Dish newDish = dishService.create(dish);
        return ResponseHandler.createResponse("Created a dish successfully", HttpStatus.OK, newDish);
    }

    @PostMapping("/")

    public ResponseEntity<Object> delete(@RequestBody Long id) {
        Boolean didDelete = dishService.delete(id);
        return ResponseHandler.createResponse("Deleted a dish successfully", HttpStatus.OK, didDelete);
    }

    @PostMapping("/")

    public ResponseEntity<Object> update(@RequestBody Dish dish) {
        Dish updatedDish = dishService.update(dish);
        return ResponseHandler.createResponse("Updated a dish successfully", HttpStatus.OK, updatedDish);
    }
}

File: Android/app/src/main/java/com/example/finalassessment/controllers/DishController.kt
package com.example.finalassessment.controllers

import com.example.finalassessment.models.Dish
import com.example.finalassessment.models.DishResponse
import com.example.finalassessment.models.SingleDishResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface DishController {

    @GET("api/v1/dishes/")
    suspend fun getDishes(): Response<DishResponse>

    @POST("api/v1/dishes/")
    suspend fun createDish(@Body dish: Dish): Response<SingleDishResponse>

    @GET("api/v1/dishes/{id}")
    suspend fun getDish(@Path("id") id: Int): Response<SingleDishResponse>
}